zerver/models.py

```diff
@@ -1423,20 +1423,42 @@ def get_realm_playgrounds(realm: Realm) -> List[RealmPlaygroundDict]:
     return playgrounds
 
 
-# The Recipient table is used to map Messages to the set of users who
-# received the message.  It is implemented as a set of triples (id,
-# type_id, type). We have 3 types of recipients: Huddles (for group
-# private messages), UserProfiles (for 1:1 private messages), and
-# Streams. The recipient table maps a globally unique recipient id
-# (used by the Message table) to the type-specific unique id (the
-# stream id, user_profile id, or huddle id).
 class Recipient(models.Model):
+    """Represents an audience that can potentially receive messages in Zulip.
+
+    This table essentially functions as a generic foreign key that
+    allows Message.recipient_id to be a simple ForeignKey representing
+    the audience for a message, while supporting the different types
+    of audiences Zulip supports for a message.
+
+    Recipient has just two attributes: The enum type, and a type_id,
+    which is the ID of the UserProfile/Stream/Huddle object containing
+    all the metadata for the audience. There are 3 recipient types:
+
+    1. 1:1 private message: The type_id is the ID of the UserProfile
+       who will receive any message to this Recipient. The sender
+       of such a message is represented separately.
+    2. Stream message: The type_id is the ID of the associated Stream.
+    3. Group private message: In Zulip, group private messages are
```

Comment:
Small thing: I'd prefer to put "1:1 private message" and "Group private message" adjacent in the list, because one tends to think of them together.

No real preference on the ordering otherwise. My suggestion would be stream, then 1:1, then group, just to go from shorter descriptions to longer ones.

Reply:
(I guess this reflects the order of the enum constants PERSONAL, STREAM, HUDDLE. … And I guess one does actually encounter those values when spelunking in the database, if nothing else. So :shrug: perhaps that's a good enough reason to keep this otherwise-illogical-seeming order in this bit of documentation.)